Launch GUI without sudo or pkexec

calexandru2018 opened the pull request:
**UPDATE**
- Support CLI v.2.2.3
- Sudo/Root privileges are kept to a bare minimum
- Users can now launch the GUI with only `protonvpn-gui`
- A .policy file is no longer needed since GUI does not launch with sudo/pkexec
- If PolKit Support is enabled, then users receive a window prompt to authorize sudo changes
- If PolKit is not enabled, users will have to either add the GUI to `visudo` or launch it from terminal, 
so that sudo password is provided whenever needed.
- Privilege escalation is needed only for the following cases:
  - Update username and password (root protected file)
  - To enable/disable autoconnect (create/remove .service file)
  - To enable/disable PolKit Support (create/remove .policy file)
  - To connect/disconnect to/from VPN (run CLI commands)

**FIXES**
- Fixes #44 
- Fixes #72 
- Fixes #74

calexandru2018 committed:
Sudo priveliege is needed to update username and password
This can use either sudo or pkexec, depending on a users preference

calexandru2018 committed:
Function should not return any output if intput is bool

Addez [a-z] characeters

calexandru2018 committed:
Created new function that handles all sudo connection requests

calexandru2018 committed:
Quick connect adapated to new sudo connection requests

calexandru2018 committed:
Created new function to further abstract how message is returned to be displayed to the user

calexandru2018 committed:
Further abstracted how display message is shown

calexandru2018 committed:
Display message abstracted for selected server

calexandru2018 committed:
Display message abstracted for selected country

calexandru2018 committed:
Display message abstracted for custom quick connect

calexandru2018 committed:
Display message abstracted for random connect

calexandru2018 committed:
Display message abstracted for last connect

calexandru2018 committed:
Display message abstracted for disconnect

calexandru2018 committed:
Removed unnecessary logging

calexandru2018 committed:
Added function to handle root commands

calexandru2018 committed:
Autoconnect asks for root privilege

User will need to provide root access upon making changes

calexandru2018 committed:
Adapted method to new escalation strategy

calexandru2018 committed:
Added new property that fetches the value of sudo_type
Updated PolKit and Autoconnect manager so that it makes use of this new property
Updated also set_user_pass so it makes use of sudo_type property
Removed reload_daemon since it is no longer being used

calexandru2018 committed:
Updated to adapt new service changes

calexandru2018 committed:
Removed keep so that root privilege is not kept for X time

calexandru2018 committed:
Added new property

calexandru2018 committed:
Function now uses the dynamic property sudo_type

calexandru2018 committed:
Added new function that executes polkit root commands, and takes into consideration timeouts

calexandru2018 committed:
Updated string

calexandru2018 committed:
Refactored code

calexandru2018 committed:
Reflected changes of the v.2.2.3 of the CLI

calexandru2018 committed:
Function now makes use of root commands to display more customized messages

calexandru2018 committed:
Ordered method so that they follow a logical path

Updated string

calexandru2018 committed:
Message is now displayed to the user if passfile is not able to be created
User should launch for first time from terminal to create the passfile

calexandru2018 committed:
Entry point now does not check for root privilege

calexandru2018 committed:
Removed change owner

calexandru2018 committed:
Removed function reference in files since sudo is not more used

calexandru2018 committed:
Implemented new sudo_type property

Adapted necessary code to use this property

calexandru2018 committed:
GUI is no longer being called with sudo

calexandru2018 committed:
Updated info string

calexandru2018 committed:
Removed autoconnect since its managed by the GUI conf

calexandru2018 committed:
Fixed testing issue for updating user and pass

calexandru2018 committed:
Added loggers

calexandru2018 committed:
Commented code since pytests does not seem to accept multiple input

calexandru2018 committed:
Updated path to be relative

calexandru2018 committed:
Username was not being updated in the config file

Although all was correctly updated in the passfile

calexandru2018 committed:
Specified withi username and password should be provided

calexandru2018 committed:
Quick connect to secure-core if user enables only sc servers

calexandru2018 committed:
Updated readme to reflect new changes

calexandru2018 committed:
Updated folder name

calexandru2018 committed:
Added logger to connections

calexandru2018 committed:
Updated path to package

calexandru2018 committed:
Updated path to package, since it was updated

calexandru2018 committed:
Updated path to package

calexandru2018 committed:
Updated link to help pages

calexandru2018 committed:
Removed sudo reference from readme

kaplun commented on linux_gui/constants.py:
```diff
@@ -76,7 +76,7 @@
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
-      <allow_active>auth_admin_keep</allow_active>
+      <allow_active>auth_admin</allow_active>
```
So this is forcing the user to type password all the time right? Is there some intermediate way? E.g. to remember password for the current X session (unless screen locked or something similar?)

calexandru2018 committed:
Added loggers and added control logic

calexandru2018 committed:
Updated logger name

calexandru2018 committed:
Added getter and setter for polkit support

Since the GUI is no longer needed to run as root, there is no need for a .policy file, because the GUI will never be run with pkexec, which is a wrapper for sudo.

calexandru2018 committed:
Updated function name and removed unnecessary commented code

calexandru2018 committed:
Removed polkit_path since they are no longer needed

calexandru2018 committed:
Removed unnecesary import

calexandru2018 committed:
Updated on_load logic for polkit since getters/setters are used

Updated also function names

calexandru2018 committed:
On login, should now detect if polkit is installed or not

If PolKit is installed then the user will be automatically prompted to enter password. If PolKit is not detected then it will fallback to sudo.

calexandru2018 committed:
Commented tests since root privilege is needed and pkexec is detected, thus is executed

Since there is no way of providing root password to pkexec via GitHub actions, test were for now commented to avoid future GitHub action errors

calexandru2018 committed:
Commented pytest testing

calexandru2018 committed:
Since GUI no longer needs root, there is no need for checking it

calexandru2018 committed:
Fixes #74

calexandru2018 committed:
Corrected spelling upon enabling/disabling PolKit Support

calexandru2018 committed:
Added loggers

calexandru2018 committed:
Added loggers

calexandru2018 committed:
Added logger

calexandru2018 committed:
Specified what loggers should return

calexandru2018 committed:
Function should only add IP to list, and not enable/disable split tunneling

calexandru2018 committed:
Fixed KeyError issue

If the user had previously installed the GUI, then this key had not been created. Enabling PolKit would through a KeyError message. This has been now addressed.
